### PR TITLE
RelativeRect

### PR DESCRIPTION
**Pull request by Hixie**
Introduce a class that can describe the metrics used in a Positioned
widget, to enable easier manipulation and storage of the values, e.g.
for animation purposes.

### CODE REVIEW

File: sky/packages/sky/lib/src/rendering/stack.dart [hunk 1]
```diff
@@ -3,10 +3,111 @@
...
+/// container, this class has no width and height members. To determine the
+/// width or height of the rectangle, convert it to a [Rect] using [toRect()]
+/// (passing the container's own Rect), and then examine that object.
+class RelativeRect {
+  const RelativeRect.fromLTRB(this.left, this.top, this.right, this.bottom);
+
+  factory RelativeRect.fromRect(Rect rect, Rect container) {
+    return new RelativeRect.fromLTRB(rect.left, rect.top, container.width - rect.right, container.height - rect.bottom);
+  }
+
```
Comment (krisgiesing) [inline sky/packages/sky/lib/src/rendering/stack.dart]: Is the input rect here expected to be already in the coordinates of the container?  That is, if I have a container of (5, 5, 10, 10), and I want a relative rect that fills the whole thing, do I pass (0, 0, 5, 5) as rect?  I deduce that's true by the math.  If so, then I think it's worth explaining in the comments, as my naive expectation was that both rectangles would be in some global coordinates, and I was then surprised not to see the container's top and left accounted for in the conversion.

**Comment by krisgiesing** [comment]
LGTM

File: sky/packages/sky/lib/src/rendering/stack.dart (hunk 1 above)
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/stack.dart]: oh, good catch. I forgot that container might not be at 0,0. Will fix.

**Comment by krisgiesing** [comment]
LGTM

**Pull request merged**